pkg/httpproxy/sign.go

```diff
@@ -0,0 +1,230 @@
+package httpproxy
+
+import (
+	"bytes"
+	"crypto/md5"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"github.com/rancher/norman/httperror"
+	"github.com/sirupsen/logrus"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/signer/v4"
+	"github.com/rancher/types/apis/core/v1"
+)
+
+type Signer interface {
+	sign(*http.Request, v1.SecretInterface, string) error
+}
+
+func newSigner(auth string) Signer {
+	splitAuth := strings.Split(auth, " ")
+	switch strings.ToLower(strings.ToLower(splitAuth[0])) {
+	case "awsv4":
+		return awsv4{}
+	case "bearer":
+		return bearer{}
+	case "basic":
+		return basic{}
+	case "digest":
+		return digest{}
+	case "arbitrary":
+		return arbitrary{}
+	}
+	return nil
+}
+
+func (br bearer) sign(req *http.Request, secrets v1.SecretInterface, auth string) error {
+	data, secret, err := getAuthData(auth, secrets, []string{"passwordField", "credID"})
+	if err != nil {
+		return err
+	}
+	req.Header.Set(AuthHeader, fmt.Sprintf("%s %s", "Bearer", secret[data["passwordField"]]))
```

Comment:
Is the secret[data["passwordField"]] encoded?

Reply:
no